test/mysql.test.js

```diff
@@ -424,6 +431,101 @@ describe('mysql', function() {
+      PostWithNumId.destroyAll(done);
+    });
+    after(function(done) {
+      PostWithNumId.destroyAll(done);
+    });
+    it('should handle null in inq operator', function(done) {
+      defaultPost.id = 1;
+      defaultPost.title = 'Foo';
+      defaultPost.content = 'Bar';
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {id: {inq: [null, 1]}}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].title.should.equal('Foo');
+          posts[0].id.should.equal(1);
+          done();
+        });
+      });
+    });
+
+    it('should handle null in nin operator', function(done) {
+      defaultPost.id = 2;
+      defaultPost.title = 'Make';
+      defaultPost.content = 'Toyota';
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {id: {nin: [null, 3]}}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].content.should.equal('Toyota');
+          posts[0].id.should.equal(2);
+          done();
+        });
+      });
+    });
+
+    it('should handle null in neq operator', function(done) {
+      defaultPost.id = 3;
+      defaultPost.title = 'Model';
+      defaultPost.content = 'Corolla';
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {id: {neq: null}}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].content.should.equal('Corolla');
+          posts[0].id.should.equal(3);
+          done();
+        });
+      });
+    });
+
+    it('should handle null in nin op for different datatypes', function(done) {
+      PostWithNumId.create(defaultPost, function(err, post) {
+        should.not.exist(err);
+        post.id.should.equal(defaultPost.id);
+        PostWithNumId.find({where: {and: [
+        {id: {nin: [null]}},
+        {title: {nin: [null]}},
+        {content: {nin: [null]}},
+        {buffProp: {nin: [null]}},
+        {objProp: {nin: [null]}},
+        {arrProp: {nin: [null]}},
+        {dateProp: {nin: [null]}},
+        {pointProp: {nin: [null]}},
+        ]}}, function(err, posts) {
+          should.not.exist(err);
+          posts.length.should.equal(1);
+          posts[0].id.should.equal(defaultPost.id);
+          posts[0].title.should.equal(defaultPost.title);
+          posts[0].content.should.equal(defaultPost.content);
+          posts[0].buffProp.should.deepEqual(defaultPost.buffProp);
+          posts[0].objProp.should.deepEqual(defaultPost.objProp);
+          posts[0].pointProp.should.deepEqual(defaultPost.pointProp);
```

Comment:
@Amir-61 @kjdelisle I did that at first, but `posts[0]` has way more keys in it than `defaultPost`.
```javascript
 1) mysql null vals in different operators should handle null in nin op for different datatypes:

      Uncaught AssertionError: expected ModelConstructor {
  __cachedRelations: Object {},
  __data: Object {
    arrProp: List [ 0 ],
    buffProp: Buffer [ 64, 65, 66, 42, 75, 66, 66, 65, 72 ],
    content: 'defContent',
    dateProp: 2017-06-14 11:02:13.000 -0400,
    id: 3,
    objProp: Object { defKey: 'defVal' },
    pointProp: Object { lat: 57.2314, lng: 4.51515 },
    title: 'defTitle'
  },
  __dataSource: undefined,
  __persisted: true,
  __strict: true,
  __unknownProperties: Array []
} to equal Object {
  arrProp: Array [ 0 ],
  buffProp: Buffer [ 64, 65, 66, 42, 75, 66, 66, 65, 72 ],
  content: 'defContent',
  dateProp: 2017-06-14 11:02:12.984 -0400,
  id: 3,
  objProp: Object { defKey: 'defVal' },
  pointProp: Object { lat: 57.2314, lng: 4.51515 },
  title: 'defTitle'
}
      + expected - actual

       {
      -  "__cachedRelations": {}
      -  "__data": {
      -    "arrProp": [
      -      0
      -    ]
      -    "buffProp": [Buffer: [
      -      100
      -      101
      -      102
      -      66
      -      117
      -      102
      -      102
      -      101
      -      114
      -    ]]
      -    "content": "defContent"
      -    "dateProp": [Date: 2017-06-14T15:02:13.000Z]
      -    "id": 3
      -    "objProp": {
      -      "defKey": "defVal"
      -    }
      -    "pointProp": {
      -      "lat": 57.2314
      -      "lng": 4.51515
      -    }
      -    "title": "defTitle"
      +  "arrProp": [
      +    0
      +  ]
      +  "buffProp": [Buffer: [
      +    100
      +    101
      +    102
      +    66
      +    117
      +    102
      +    102
      +    101
      +    114
      +  ]]
      +  "content": "defContent"
      +  "dateProp": [Date: 2017-06-14T15:02:12.984Z]
      +  "id": 3
      +  "objProp": {
      +    "defKey": "defVal"
         }
      -  "__dataSource": [undefined]
      -  "__persisted": true
      -  "__strict": true
      -  "__unknownProperties": []
      +  "pointProp": {
      +    "lat": 57.2314
      +    "lng": 4.51515
      +  }
      +  "title": "defTitle"
       }
```